contracts/account/modules/ERC7579Module.sol

```diff
@@ -0,0 +1,57 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.20;
+
+import {IERC7579Module} from "@openzeppelin/contracts/interfaces/draft-IERC7579.sol";
+
+/**
+ * @dev Abstract contract for implementing ERC-7579 modules.
+ *
+ * This contract provides a basic implementation of the ERC-7579 module interface,
+ * which allows for modules to be installed and uninstalled from an ERC-7579 account.
+ *
+ * Developers must specify the module type ID in the constructor and `implement the {onInstall} and {onUninstall} functions in derived contracts.
+ *
+ * Example usage:
+ *
+ * ```solidity
+ * contract MyExecutorModule is ERC7579Module(MODULE_TYPE_EXECUTOR) {
+ *     function onInstall(bytes calldata data) public override {
+ *         // Install logic here
+ *     }
+ *
+ *     function onUninstall(bytes calldata data) public override {
+ *         // Uninstall logic here
+ *     }
+ *
+ * }
+ * ```
+ */
+abstract contract ERC7579Module is IERC7579Module {
+    uint256 private immutable _moduleTypeId;
+
+    /// @dev Emitted when a module is installed on an account.
+    event ModuleInstalledReceived(address account, bytes data);
+
+    /// @dev Emitted when a module is uninstalled from an account.
+    event ModuleUninstalledReceived(address account, bytes data);
+
+    constructor(uint256 moduleTypeId) {
+        _moduleTypeId = moduleTypeId;
+    }
```

Comment:
I don't like that this is forcing an immutable variable for `_moduleTypeId`, I think it's too opinionated and instead, it should be defined in a constant and overrideable with a virtual function